test/ApplicationFactory.ts

```diff
@@ -1,10 +1,67 @@
 import { Container } from 'inversify';
-import { Application } from 'express';
-import { InversifyExpressServer } from 'inversify-express-utils';
-import { getExpressAppConfig } from '../src/utils/ConfigLoader'
+import { Application, NextFunction, RequestHandler, Request, Response } from 'express';
+import { InversifyExpressServer, BaseMiddleware } from 'inversify-express-utils';
+import { getExpressAppConfig, loadEnvironmentVariablesFromFiles } from '../src/utils/ConfigLoader';
+import { AuthMiddleware } from '../src/middleware/AuthMiddleware';
+import { getEnvOrDefault } from '../src/utils/EnvironmentUtils';
+import { Maybe, SessionStore, EitherUtils, SessionMiddleware, Session } from 'ch-node-session-handler';
+import { Cookie } from 'ch-node-session-handler/lib/session/model/Cookie';
+import Substitute from '@fluffy-spoon/substitute';
 
-export const createApplication = (configureContainerBindings: (container: Container) => void = () => {}): Application => {
+export const createAppConfigurable = (configureContainerBindings: (container: Container) => void = () => { }): Application => {
+
+    loadEnvironmentVariablesFromFiles();
     const container = new Container();
     configureContainerBindings(container);
     return new InversifyExpressServer(container).setConfig(getExpressAppConfig('../../')).build();
 };
+
+export const setupFakeAuth = (container: Container) => {
+    const fakeHandler = (req: Request, res: Response, next: NextFunction) => {
+        return next();
+    };
+
+    class FakeMiddleware extends AuthMiddleware {
+        handler: RequestHandler = fakeHandler;
+    }
+
+    const newFakeMiddleware = new FakeMiddleware();
+    container.bind<AuthMiddleware>(AuthMiddleware).toConstantValue(newFakeMiddleware);
+};
+
+export const getDefaultConfig = () => {
+    loadEnvironmentVariablesFromFiles();
+    return {
+        cookieName: getEnvOrDefault('COOKIE_NAME'),
+        cookieSecret: getEnvOrDefault('COOKIE_SECRET')
+    };
+};
+
+export const createApp = (session?: Session, setCookie: boolean = true) => createAppConfigurable(container => {
+
+    const config = getDefaultConfig();
+
+    const cookie = session ? Cookie.createFrom(session) : null;
+
+    const sessionStore = Substitute.for<SessionStore>();
+
+    if (session && cookie) {
+        sessionStore.load(cookie).returns(EitherUtils.wrapValue(session.data));
+    }
+
+    const realMiddleware = SessionMiddleware(config, sessionStore);
+    const sessionHandler = (req: Request, res: Response, next: NextFunction) => {
+        req.session = session ? Maybe.of(session) : Maybe.empty();
```

Comment:
That seems to be not needed since real middleware will take care of it. Please remove.